idb/postgres/internal/writer/writer.go

```diff
@@ -364,7 +364,30 @@ func (w *Writer) addTransactionParticipation(block *bookkeeping.Block) error {
 	return nil
 }
 
-func writeAccountData(round basics.Round, address basics.Address, accountData basics.AccountData, batch *pgx.Batch) {
+type optionalSigType struct {
```

Comment:
This type isn't needed, a `map[Address]SigType` can be used. On a rekey you could `delete(res, payset[i].Txn.Sender)`, the present check would change to compare sigtype to the zero value (or check if it is in the map).

Reply:
I think that wouldn't work. We need to distinguish between three operations: leave sigtype as is, clear it, or set to something non-null. If an address is not in `map[basics.Address]optionalSigType`, we leave sigtype as is. If it is in the map and the value is `optionalSigType` with `present = false`, we clear it. Otherwise, we set sigtype to `optionalSigType.value`.

Reply:
You aren't using the 3rd case, this code sets something for each sender in the payset, even if it doesn't change.

Reply:
you mean the first case?